config/README.md

```diff
@@ -0,0 +1,77 @@
+   1. `validate` - validate tag is used to perform validation on field structs using the [validator](https://github.com/go-playground/validator) package.
+   2. `mapstructure` - mapstructure tag is used for unmarshalling and must match the CLI flag name defined in step or else the field will not be set when the config is unmarshalled.
+   ```go
+        type MyComponentConfig struct {
+            AppWorkers int `validate:"gt=0" mapstructure:"app-workers"`
+        }
+    ```
+   It's important to make sure that the CLI flag name matches the mapstructure field tag to avoid parsing errors.
+4. Add the new config and a default value to the default-config.yml file. Ensure that the new property added matches the configuration struct structure for the subpackage the config belongs to.
+    ```yaml
+      config-file: "./default-config.yml"
+      network-config:
+      ...
+      my-component:
+        app-workers: 1
+    ```
+5. Finally, if a new struct was created add it as a new field to the FlowConfig. In the previous steps we added a new config struct and added a new property to the default-config.yml for this struct `my-component`. This property name
+    must match the mapstructure field tag for the struct when added to the FlowConfig.
+    ```go
+    // FlowConfig Flow configuration.
+    type FlowConfig struct {
+        ConfigFile    string          `validate:"filepath" mapstructure:"config-file"`
+        NetworkConfig *network.Config `mapstructure:"network-config"`
+        MyComponentConfig *mypackage.MyComponentConfig `mapstructure:"my-component"`
+    }
+    ```
+
+### Nested structs
+In an effort to keep the configuration yaml structure readable some configuration will be in nested properties. When this is the case the mapstructure `squash` tag can be used so that the corresponding nested struct will be 
+flattened before the configuration is unmarshalled. This is used in the network package which is a collection of configuration structs nested on the network.Config struct. 
+```go
+type Config struct {
+    // UnicastRateLimitersConfig configuration for all unicast rate limiters.
+    UnicastRateLimitersConfig `mapstructure:",squash"`
+    ...
+}
+```
+UnicastRateLimitersConfig is a nested struct that defines configuration for unicast rate limiter component. In our configuration yaml structure you will see that all network configs are defined under the network-config property.
+
+### Setting Aliases
+Most configs will not be defined on the top layer FlowConfig but instead be defined on nested structs and in nested properties of the configuration yaml. When the default config is initially loaded the underlying config [viper](https://github.com/spf13/viper) store will store 
+each configuration with a key that is prefixed which each parent property. For example because network-connection-pruning is found on the network-config property of the configuration yaml the key used by the config store to 
```

Comment:
```suggestion
each configuration with a key that is prefixed with each parent property. For example, because `network-connection-pruning` is found on the `network-config` property of the configuration yaml, the key used by the config store to 
```